webapp/src/components/Board/BoardComponent.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import PieceComponent from './PieceComponent';
+import BoardContext from './BoardContext';
+
+const BoardComponent = ({ chess, view, lobbyID }) => {
+	const names = {
+		k: 'king',
+		n: 'knight',
+		r: 'rook',
+		q: 'queen',
+		p: 'pawn',
+		b: 'bishop',
+	};
+
+	const turn = String(chess.turn()) === 'b' ? 'black' : 'white';
+
+	const field = chess.board();
+
+	const Board = [];
+
+	field.forEach((row) => {
+		[...row]
+			.map((elem) => {
+				const tmp = {
+					skin: 'default',
+					color: elem === null ? null : elem.color === 'b' ? 'black' : 'white',
+					name: elem === null ? null : names[elem.type],
+					selected: false,
+					targeted: false,
+					idle: false,
+				};
+				return { ...tmp };
+			})
+			.forEach((dict) => {
+				Board.push({ ...dict });
```

Comment:
Could be written as
`field.forEach(row => Board.push(row.map(it => ({ selected: false, targeted: false /* row element */})))`